sql/core/src/main/scala/org/apache/spark/sql/execution/joins/CartesianProduct.scala

```diff
@@ -34,7 +34,15 @@ case class CartesianProduct(left: SparkPlan, right: SparkPlan) extends BinaryNod
     val leftResults = left.execute().map(_.copy())
     val rightResults = right.execute().map(_.copy())
 
-    leftResults.cartesian(rightResults).mapPartitions { iter =>
+    val cartesianRdd = if (leftResults.partitions.size > rightResults.partitions.size) {
+      rightResults.cartesian(leftResults).mapPartitions { iter =>
+        iter.map(tuple => (tuple._2, tuple._1))
+      }
+    } else {
+      leftResults.cartesian(rightResults)
+    }
+
+    cartesianRdd.mapPartitions { iter =>
       val joinedRow = new JoinedRow
```

Comment:
yes, use partition size here is not accurate, see a rdd with 100 partitions, and each partition has one record and a rdd with 10 partition and each partition has 100 million records, use the method above will cause more scan from hdfs